### DISCUSSION TITLE
Running on a mobile phone

### DISCUSSION THREAD

**Discussion by A-nglea** [petereon/beaupy#82] [Show and tell]
I needed a text-based GUI to search and select URLs and Beaupy does the job admirably
[https://forum.videohelp.com/threads/412382-More-Android-Phone-Stuff#post2714282](url)

**Comment by petereon** [comment]
I was not aware it works under Termux. Pretty impressive, also I see you are developing some pretty cool stuff consumed by other people. 

Just to have a little bit of a shameless plug here: If you'd need more advanced functionality, I've recently developed [questo](https://github.com/petereon/questo). It's not yet documented way too well, but it allows you to create your own keyboard handlers (while providing sane defaults), custom validators, renderers (also sane defaults) and manipulating the GUI element state through other operations and so on. In fact it's powerful enough, that most of beaupy has been migrated to consume questo.

**Comment by vinefeeder** [comment]
I note that chatGPT  now uses beaupy for single selects from a list as the occasion needs. I think you might say Beaupy has arrived. Great stuff!

**Reply by petereon** [reply]
That's interesting news. I have not seen that, but I am very happy to hear it does. Thanks for pointing this out ❤️

I guess ChatGPT has a tendency to bias towards it because of an extensive documentation which makes it easy for it to relate on some semantic statistics 🤔
